ibis/backends/base/sql/alchemy/client.py

```diff
@@ -329,3 +329,120 @@ def _sqla_table_to_expr(self, table):
+            from_table_name (table name to insert data from)
+
+            Sorry, can't insert from both the data_obj (dataframe)
+            and the from_table_name (table). Please use only one
+            parameter.
+
+        RuntimeError
+            The table already exists
+
+        TypeError
+            No operation is being performed. Either the data_obj
+            parameter is not a pandas dataframe or the
+            from_table_name parameter is not of string datatype.
+
+        """
+
+        if data_obj is None and from_table_name is None:
+            raise ValueError(
+                'You must pass either data_obj (pandas Dataframe)'
+                ' or from_table_name (table name to insert data from)'
+            )
+
+        if isinstance(data_obj, pd.DataFrame) and isinstance(
+            from_table_name, str
+        ):
+            raise ValueError(
+                "Sorry, can't insert from both the data_obj (dataframe)"
+                " and the from_table_name (table). Please use only one"
+                " parameter."
+            )
+
+        params = {}
+        if database is None:
+            params['schema'] = self.database_name
+            database = self.database_name
+
+        if isinstance(data_obj, pd.DataFrame):
+            data_obj.to_sql(
+                to_table_name,
+                self.con,
+                index=False,
+                if_exists=if_exists,
+                **params,
+            )
+        elif isinstance(from_table_name, str):
+            if if_exists == 'fail' and to_table_name in self.list_tables():
+                raise RuntimeError('The table already exists')
+            elif (
+                if_exists in ['replace', 'append']
+            ) and to_table_name in self.list_tables():
+                to_table_expr = self.table(to_table_name)
+                to_table_schema = to_table_expr.schema()
+
+                if if_exists == 'replace':
+                    self.drop_table(to_table_name, database=database)
+                    self.create_table(
+                        to_table_name,
+                        schema=to_table_schema,
+                        database=database,
+                    )
+
+                to_table = self._table_from_schema(
+                    to_table_name,
+                    to_table_schema,
+                    database=database or self.current_database,
+                )
+
+                from_table_expr = self.table(from_table_name)
+
+                with self.begin() as bind:
+                    to_table.create(bind=bind)
+                    if from_table_expr is not None:
+                        bind.execute(
+                            to_table.insert().from_select(
+                                list(from_table_expr.columns),
+                                from_table_expr.compile(),
+                            )
+                        )
+        else:
+            raise TypeError(
```

Comment:
why is this a TypeError?

Reply:
Because we are expecting `from_table_name` to be of type `str` and if anything else was given we wanted to raise an error (based on discussion during previous reviews).